### PR TITLE
Clean up some workarounds to the release process

### PR DESCRIPTION
**Pull request by scotthain**
This removes some workarounds regarding promotion of artifacts with different package targets. This requires 0.80.0.

### CODE REVIEW

File: .buildkite/scripts/build_component.sh [hunk 1]
```diff
@@ -41,37 +41,15 @@ else
...
-
-promote "${pkg_ident}" "${pkg_target}" "${channel}"
-set_target_metadata "${pkg_ident}" "${pkg_target}"
+echo "--- :habicat: Uploading ${pkg_ident} to Builder in the '${channel}' channel"
+${hab_binary} pkg upload \
+    --channel="${channel}" \
+    --auth="${HAB_AUTH_TOKEN}" \
+    "results/${pkg_artifact}"
+
+${hab_binary} pkg promote \
```
Comment (smacfarlane) [inline .buildkite/scripts/build_component.sh]: I know this is an uncomment of previous behavior,  but I thought `hab pkg upload --channel` promoted the package to the channel on upload.  Is there a reason for this promote that I'm missing context on?

File: .buildkite/scripts/build_component.sh [hunk 2]
```diff
@@ -41,37 +41,15 @@ else
...
-     --header "Content-Type: application/octet-stream" \
-     --header "Authorization: Bearer $HAB_AUTH_TOKEN" \
-     --data-binary "@results/${pkg_artifact:?}" \
-     --fail \
-     --verbose \
-    "https://bldr.habitat.sh/v1/depot/pkgs/${pkg_ident}?checksum=${pkg_blake2bsum:?}&target=${pkg_target}"
-
-promote "${pkg_ident}" "${pkg_target}" "${channel}"
-set_target_metadata "${pkg_ident}" "${pkg_target}"
+echo "--- :habicat: Uploading ${pkg_ident} to Builder in the '${channel}' channel"
```
Comment (smacfarlane) [inline .buildkite/scripts/build_component.sh]: should be `"--- :habicat: :partyparrot:`

File: .buildkite/scripts/build_component.sh (hunk 1 above)
Comment (christophermaier) [reply .buildkite/scripts/build_component.sh]: We had to do them separately when using `curl`, but before that workaround, we uploaded straight to the channel using `hab`.

I can't think of a reason to not go back to that.

Comment (scotthain) [reply .buildkite/scripts/build_component.sh]: Fixed!

File: .buildkite/scripts/build_component.sh [hunk 3]
```diff
@@ -41,37 +41,11 @@ else
...
+${hab_binary} pkg upload \
+    --channel="${channel}" \
+    --auth="${HAB_AUTH_TOKEN}" \
+    "results/${pkg_artifact}"
```
Comment (christophermaier) [inline .buildkite/scripts/build_component.sh]: We should double-check that this behaves properly with respect to package target before getting this merged. We can do a "fake release", as detailed in https://github.com/habitat-sh/habitat/pull/6510.

**Comment by baumanj** [comment]
What's the next step here?

**Comment by scotthain** [comment]
Needs to be updated - I ran it through a fake release and it had some issues. I'll resolve them.

**Comment by smacfarlane** [comment]
This work has been incorporated into #6601.  Thanks @scotthain!

**Pull request closed without merging**